spire/public/api.py

```diff
@@ -143,22 +152,117 @@ async def get_public_journal_handler(
 )
 async def get_public_journal_entries_handler(
     journal_id: UUID = Path(...),
-    db_session: Session = Depends(db.yield_connection_from_env),
+    db_session: Session = Depends(yield_db_read_only_session),
 ) -> BugoutJournalEntries:
     """
     Get list of public journal entries.
 
     - **journal_id** (uuid): Journal ID
     """
     try:
-        public_journal = await actions.get_public_journal(db_session, journal_id)
-        public_user = await actions.get_public_user(db_session, public_journal.user_id)
+        public_user = actions.get_public_journal_user(
+            db_session=db_session, journal_id=journal_id
+        )
+
+        result = bugout_api.get_entries(
+            token=public_user.restricted_token_id, journal_id=journal_id
+        )
     except actions.PublicJournalNotFound:
         raise HTTPException(status_code=404, detail="Public journal not found")
+    except BugoutResponseException as e:
+        raise HTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(e)
+        raise HTTPException(status_code=500)
+
+    return result
+
+
+@app_public.post(
+    "/{journal_id}/entries", tags=["public journals"], response_model=BugoutJournalEntry
+)
+async def create_public_journal_entry_handler(
+    journal_id: UUID = Path(...),
+    create_request: BugoutJournalEntryRequest = Body(...),
+    db_session: Session = Depends(yield_db_read_only_session),
+) -> BugoutJournalEntry:
+    """
+    Create public journal entry as autogenerated user.
+    """
+    for f in actions.entry_fields_length_limit:
+        req_attr = getattr(create_request, f["name"])
+        if req_attr is not None:
+            if len(req_attr) > f["max_length"]:
+                raise HTTPException(
+                    status_code=413,
+                    detail=f"Field {f['name']} length too long for public access",
+                )
+
+    for tag in create_request.tags:
+        if len(tag) > 50:
+            raise HTTPException(
+                status_code=413, detail="Tag length too long for public access"
+            )
+    try:
+        public_user = actions.get_public_journal_user(
```

Comment:
@kompotkot : Can anyone create entries in any public journal? What if we remove `journal.entries.create` permission from public user on a public journal?

Maybe we should do permission check? And maybe we should take argument of permissions for public user when we create that user for a public journal?

Reply:
Yes, anyone can create entries in public journal if such permissions granted to public user with cli. If we will remove `journal.entries.create` permission, it means that user became just a read-only user. There is no consistency in database which we need to check, we are duplicate what permissions we granted to user.

I think we can add endpoint to check what kind of permissions do we have according to some journal.